packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -0,0 +1,825 @@
+    `link[rel="preload"][href="${limitedEscapedHref}"]`,
+  );
+  if (!element) {
+    element = createResourceInstance('link', props, ownerDocument);
+    insertPreloadInstance(element, ownerDocument);
+  }
+  return {
+    type: 'preload',
+    href: href,
+    ownerDocument,
+    props,
+    instance: element,
+  };
+}
+
+function acquireStyleResource(resource: StyleResource): Instance {
+  if (!resource.instance) {
+    const {props, ownerDocument, precedence} = resource;
+    const limitedEscapedHref = escapeSelectorAttributeValueInsideDoubleQuotes(
+      props.href,
+    );
+    const existingEl = ownerDocument.querySelector(
+      `link[rel="stylesheet"][data-rprec][href="${limitedEscapedHref}"]`,
+    );
+    if (existingEl) {
+      resource.instance = existingEl;
+      resource.preloaded = true;
+      const loadingState: ?StyleResourceLoadingState = (existingEl: any)._p;
+      if (loadingState) {
+        // if an existingEl is found there should always be a loadingState because if
+        // the resource was flushed in the head it should have already been found when
+        // the resource was first created. Still defensively we gate this
+        switch (loadingState.s) {
+          case 'l': {
+            resource.loaded = true;
+            resource.error = false;
+            break;
+          }
+          case 'e': {
+            resource.error = true;
+            break;
+          }
+          default: {
+            attachLoadListeners(existingEl, resource);
+          }
+        }
+      } else {
+        resource.loaded = true;
+      }
+    } else {
+      const instance = createResourceInstance(
+        'link',
+        resource.props,
+        ownerDocument,
+      );
+
+      attachLoadListeners(instance, resource);
+      insertStyleInstance(instance, precedence, ownerDocument);
+      resource.instance = instance;
+    }
+  }
+  resource.count++;
+  return resource.instance;
+}
+
+function attachLoadListeners(instance: Instance, resource: StyleResource) {
+  const listeners = {};
+  listeners.load = onResourceLoad.bind(
+    null,
+    instance,
+    resource,
+    listeners,
+    loadAndErrorEventListenerOptions,
+  );
+  listeners.error = onResourceError.bind(
+    null,
+    instance,
+    resource,
+    listeners,
+    loadAndErrorEventListenerOptions,
+  );
+
+  instance.addEventListener(
+    'load',
+    listeners.load,
+    loadAndErrorEventListenerOptions,
+  );
+  instance.addEventListener(
+    'error',
+    listeners.error,
+    loadAndErrorEventListenerOptions,
+  );
+}
+
+const loadAndErrorEventListenerOptions = {
+  passive: true,
+};
+
+function onResourceLoad(
+  instance: Instance,
+  resource: StyleResource,
+  listeners: {[string]: () => mixed},
+  listenerOptions: typeof loadAndErrorEventListenerOptions,
+) {
+  resource.loaded = true;
+  resource.error = false;
+  for (const event in listeners) {
+    instance.removeEventListener(event, listeners[event], listenerOptions);
```

Comment:
I don't know if we need to bother removing these. We can instead just add a closure and clean up any heavy references within the closure.

Reply:
We don't clean up the `_p` thing anyway.

Reply:
Not sure I follow. I can just leave the listeners in place but what references are heavy?